cluster.go

```diff
@@ -176,14 +179,15 @@ func (c *ClusterClient) resetClients() (err error) {
 func (c *ClusterClient) setSlots(slots []ClusterSlotInfo) {
 	c.slotsMx.Lock()
 
-	c.slots = make([][]string, hashSlots)
 	c.resetClients()
-
 	seen := make(map[string]struct{})
 	for _, addr := range c.addrs {
 		seen[addr] = struct{}{}
 	}
 
+	for i := 0; i < hashSlots; i++ {
```

Comment:
This should be much more slower than `make([][]string, hashSlots)`. But I don't insist.